### ISSUE TITLE
Application data not preserved after android device disconnected and then reconnected to USB

### ISSUE THREAD

**Issue by ish-1313** [xamarin/Xamarin.Forms#4602]
### Description
first time after android device connected to USB, application will be completely uninstalled and data, created on the device will be lost

### Steps to Reproduce

1. debug application on the device and put some data Environment.SpecialFolder.Personal
2. close application, disconnect device from usb 
3. reconnect device to usb, start debugging, check that data was cleared from device

checked this with adb. even if I copy my data (SQLite database ) to different folders in com.companyname.applicationame - it will be removed

### Expected Behavior
data should be preserved 
or
data should be deleted or preserved with respect to "preserve application data cache..." checkbox in options

### Actual Behavior
data deleted
### Basic Information

- Version with issue: 3.4.0.1008975
- Last known good version: unknown
- IDE: visual studio 15.9.2
- Platform Target Frameworks: 
  - Android:  8.0
- Affected Devices: Samsung S7 edge with oreo

Windows 10.0.17134.407 running on Dell XPS15 9550

**Comment by kingces95** [comment]
So this misbehavior happens at design/develop time? Typically, restarting debugging involves re-installing the app which might reasonably be expected to clear any existing data, no?

**Comment by ish-1313** [comment]
yes in design/develop time. in my opinion it is unexpected and very inconvenient.

In my situation, debugging application which stores all user data in local database every time I need to start from scratch with data or use different approach like saving data outside of application space . the same problem arise when you install app on somebody else device for testing.

I think the correct behavior should be the same as updating from play market. this also allows to debug internal update process (like changing db scheme, converting other program data)

well, this is my first xamarin.forms app and first app for android, but I have no problems with local data with apps for windows phone or ios. I assume, that android apps updates from play market do not delete app data

**Comment by kvpt** [comment]
It's a parameter in Visual Studio, see the capture on the microsoft documentation :
https://docs.microsoft.com/en-us/xamarin/android/troubleshooting/questions/android-sdk-location?tabs=windows

**Comment by ish-1313** [comment]
no, this behavior doesn't respect this checkbox
